### PR TITLE
fix: any-all of fix

### PR DESCRIPTION
**Pull request by dimakis**
## Description

fixes an issue where openapi generated SDKs can not get list of connectors

fix has already been tested in SDK and works
## Checklist (Definition of Done)

- [ ] All acceptance criteria specified in JIRA have been completed
- [ ] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- [ ] Documentation added for the feature
- [ ] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] All PR comments are resolved either by addressing them or creating follow up tasks
- [ ] Required metrics/dashboards/alerts have been added (or PR created).
- [ ] Required Standard Operating Procedure (SOP) is added.
- [ ] JIRA has been created for changes required on the client side

### CODE REVIEW

File: openapi/connector_mgmt.yaml [hunk 1]
```diff
@@ -1267,15 +1267,9 @@ components:
     ConnectorType:
       description: >-
         Represents a connector type supported by the API
```
Comment (VanillaSpoon) [inline openapi/connector_mgmt.yaml:1269]: Could remove this split as there's only one line in the description :)
```diff
-      description: >-
-        Represents a connector type supported by the API
+     description: Represents a connector type supported by the API
```

**Review by VanillaSpoon — COMMENTED** [review]
```diff
          {
            "href": "/api/connector_mgmt/v1/kafka_connector_types/OldConnectorTypeStillInUseId",
            "id": "OldConnectorTypeStillInUseId",
            "kind": "ConnectorType",
            "name": "Old Connector Type still in use",
+           "version": ""
          }

```
Add version [Here](https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/internal/connector/test/integration/features/connector-api.feature#L485) to resolve the CI failing Issues :)

**Review by valdar — CHANGES_REQUESTED** [review]

File: internal/connector/internal/api/admin/private/api/openapi.yaml [hunk 2]
```diff
@@ -2061,15 +2061,9 @@ components:
...
       description: Represents a connector type supported by the API
-      oneOf:
-      - required:
-        - name
-        - schema
-        - version
-      - required:
-        - json_schema
-        - name
-        - version
```
[the remark this answers is not in the document]
Comment (dimakis) [reply internal/connector/internal/api/admin/private/api/openapi.yaml:2072]: that's grand. I've got it updated now. thank you

File: internal/connector/internal/api/public/api/openapi.yaml [hunk 3]
```diff
@@ -1895,15 +1895,9 @@ components:
...
       description: Represents a connector type supported by the API
-      oneOf:
-      - required:
-        - name
-        - schema
-        - version
-      - required:
-        - json_schema
-        - name
-        - version
```
[the remark this answers is not in the document]
Comment (dimakis) [reply internal/connector/internal/api/public/api/openapi.yaml:1906]: 👍

File: openapi/connector_mgmt.yaml [hunk 4]
```diff
@@ -1282,17 +1282,10 @@ components:
...
-        Represents a connector type supported by the API
-      oneOf:
-        - required:
-            - name
-            - version
-            - schema
-        - required:
-            - name
-            - version
-            - json_schema
```
[the remark this answers is not in the document]
Comment (dimakis) [reply openapi/connector_mgmt.yaml:1295]: 👍

File: internal/connector/internal/api/admin/private/api/openapi.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (valdar) [reply internal/connector/internal/api/admin/private/api/openapi.yaml:2072]: @dimakis the name to be kept should be `schema` not `json_schema`. I think you should remove any reference to `json_schema`  in the properties as well: https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/f259d085137bb4ff3fe7dfecc4f51320992984df/openapi/connector_mgmt.yaml#L1315

Beside that I think there will be test failures... I think I can take care of those once you finish the changes.

[the remark this answers is not in the document]
Comment (dimakis) [reply internal/connector/internal/api/admin/private/api/openapi.yaml:2072]: thanks @valdar, I've updated this and removed json_schema

**Comment by valdar** [comment]
@dimakis if you resolve conflicts I think this is good to be merged

**Review by valdar — APPROVED** [review]

**Review by dhirajsb — COMMENTED** [review]
@gashcrumb can you check whether the UI has any old references still to `json_schema` anywhere? It probably doesn't, but doesn't hurt to check.

**Comment by gashcrumb** [comment]
@dhirajsb yep, took a look, besides some test fixtures that need to be updated we don't use that property.

**Review by dhirajsb — APPROVED** [review]
lgtm

**Pull request merged**